### PR TITLE
020 delete role

### PR DESCRIPTION
**Pull request by conormcgee480**
https://trello.com/c/Hij2Lil1

### CODE REVIEW

**Comment by conormcgee480** [comment]
[US020 - Delete a role (3)](https://trello.com/c/Hij2Lil1/20-us020-delete-a-role-3)

Reviewed comments I (Jennifer) made have been changed in later commit below forgot to add the comments to this pull request

**Review by jennifer-choo — APPROVED** [review]

File: src/main/java/com/kainos/ea/controller/RecruitmentRequests.java [hunk 1]
```diff
@@ -7,9 +7,7 @@
 
 import org.eclipse.jetty.http.HttpStatus;
 
-import javax.ws.rs.GET;
-import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
+import javax.ws.rs.*;
```
Comment (boydstepnew) [inline src/main/java/com/kainos/ea/controller/RecruitmentRequests.java:10]: Avoid *, might need to change your intellij settings

File: src/main/java/com/kainos/ea/dao/JobsDao.java [hunk 2]
```diff
@@ -12,8 +12,8 @@
 public class JobsDao {
     public List<Job> getJobs(Connection c) throws SQLException, DatabaseConnectionException {
         Statement st = c.createStatement();
-        PreparedStatement statement = c.prepareStatement("SELECT id, job_roles.name, description, specification, responsibilities, bands.name, capability " +
-                "FROM job_roles JOIN bands USING(id)");
+        PreparedStatement statement = c.prepareStatement("SELECT job_roles.id, job_roles.name, description, specification, responsibilities, bands.name, capability " +
+                "FROM job_roles JOIN bands ON job_roles.band_id = bands.id");
```
Comment (boydstepnew) [inline src/main/java/com/kainos/ea/dao/JobsDao.java]: Did we agree to bring this in on this ticket?

File: src/test/com/kainos/ea/integration/RecruitmentRequestsIntegrationTest.java [hunk 3]
```diff
@@ -28,4 +38,36 @@ void getJobs_shouldReturnListOfJobs() {
 
         Assertions.assertTrue(response.size() > 0);
     }
+    @Test
+    void deleteJobRole_shouldReturnJobSuccessfullyDeletedMessage() throws SQLException, DatabaseConnectionException {
+        Job job =new Job(
+                "Marketing Operations Manager",
+                "Develops Software for Kainos",
+                "https://kainossoftwareltd.sharepoint.com/:b:/g/people/EbTM1UOLa0VBvOttkOL3ZNoB0sMjehxvkAaNQEj2dqKMbA?e=iXpeUf",
```
Comment (boydstepnew) [inline src/test/com/kainos/ea/integration/RecruitmentRequestsIntegrationTest.java:46]: You can shorten this test data to keep it readable

File: src/main/java/com/kainos/ea/dao/JobsDao.java (hunk 2 above)
Comment (conormcgee480) [reply src/main/java/com/kainos/ea/dao/JobsDao.java]: Ah yes I forgot it was being moved to another adhoc ticket I'll revert it here now

File: src/test/com/kainos/ea/integration/RecruitmentRequestsIntegrationTest.java [hunk 4]
```diff
@@ -28,4 +38,36 @@ void getJobs_shouldReturnListOfJobs() {
...
+                "Experience of building and testing modern software applications",
+                6,
+                "Business Development and Marketing");
+        DatabaseConnector databaseConnector = new DatabaseConnector();
+        Connection c =databaseConnector.getConnection();
+        String sql = "INSERT INTO job_roles (name,description,specification,responsibilities,capability,band_id) VALUES(?,?,?,?,?,?);";
+        PreparedStatement InsertStatement = c.prepareStatement(sql,Statement.RETURN_GENERATED_KEYS);
```
Comment (boydstepnew) [inline src/test/com/kainos/ea/integration/RecruitmentRequestsIntegrationTest.java:85]: Could you separate this into it's own function? Would be cleaner to remove it from here

File: src/test/com/kainos/ea/integration/RecruitmentRequestsIntegrationTest.java [hunk 5]
```diff
@@ -28,4 +38,36 @@ void getJobs_shouldReturnListOfJobs() {
...
+        int insertedJobID=0;
+        if (rs.next()){
+            insertedJobID = rs.getInt(1);
+        }
+        String actualResponse = APP.client().target("http://localhost:8080/api/delete-job/"+insertedJobID)
+                .request()
+                .delete(String.class);
+        String expectedResponse = "This job role has been deleted: ID: "+insertedJobID+", Name: "+job.getName();
+        Assertions.assertEquals(actualResponse,expectedResponse);
+    }
```
Comment (boydstepnew) [inline src/test/com/kainos/ea/integration/RecruitmentRequestsIntegrationTest.java:96]: How about a test for when the job doesn't exist?

File: src/test/com/kainos/ea/service/JobsRequestServiceTest.java [hunk 6]
```diff
@@ -56,4 +56,61 @@ void getJobs_shouldThrowSqlException_whenDaoThrowsSqlException() throws SQLExcep
...
                 () -> jobsRequestService.getJobs());
     }
+
+    @Test
+    void deleteJobRoleShouldReturnCorrectDeleteMessageWhenDaoReturnsCorrectDeleteMessage() throws DatabaseConnectionException, SQLException {
+        Job job =new Job(
+                1,
+                "Software Engineer",
+                "Develops Software for Kainos",
+                "https://kainossoftwareltd.sharepoint.com/:b:/g/people/EbTM1UOLa0VBvOttkOL3ZNoB0sMjehxvkAaNQEj2dqKMbA?e=iXpeUf",
```
Comment (boydstepnew) [inline src/test/com/kainos/ea/service/JobsRequestServiceTest.java:66]: Can shorten this too :)

**Review by boydstepnew — APPROVED** [review]
